client/src/app/user.service.ts

```diff
@@ -152,7 +133,13 @@ export class UserService {
    * @param attribute (optional)
    */
   getProfile(attribute?: string) {
-    return this.http.get<UserProfile>(`${this.uri}/`, this.getHttpHeaders());
+    return this.http
+      .get<UserProfile>(`${this.uri}/`, this.getHttpHeaders())
+      .pipe(
+        catchError((err) => {
```

Comment:
I didn't know you could define catches before calling subscribe. Very cool. I was looking for something like this.